app/src/lib/rules/case_frame/common.js

```diff
@@ -343,11 +343,17 @@ export function* validate_case_frame(trigger_context) {
 	const selected_result = token.lookup_results[0]
 	const case_frame = selected_result.case_frame
 
+	if (!case_frame.is_valid) {
+		yield { error: 'Invalid argument structure for {sense}. Consult the Ontology for correct usage.' }
```

Comment:
I did not realize you were using generator functions... I've never used them, but I'd love to learn more about their utility for what you're doing.

Reply:
explained while pairing.